File: cloud-computing/src/models/user.model.js
import mongoose from 'mongoose';

const UserSchema = new mongoose.Schema(
	{
		first_name: {
			type: String,
			required: [true, 'Firstname field is required!'],
		},
		last_name: {
			type: String,
			required: [true, 'Lastname field is required!'],
		},
		email: {
			type: String,
			required: [true, 'Email field is required!'],
			unique: true,
			lowercase: true,
		},
		phone_number: {
			type: String,
			required: [true, 'Phone Number field is required!'],
			unique: true,
		},
		password: {
			type: String,
			required: [true, 'Password field is required!'],
			minlength: [8, 'Password must have at least 8 characters!'],
		},
		gender: {
			type: String,
			required: [true, 'Gender field is required!'],
		},
		address: {
			type: String,
			required: [true, 'Address field is required!'],
		},
		city: {
			type: String,
			required: [true, 'City field is required!'],
		},
	},
	{
		timestamps: {
			createdAt: 'created_at',
			updatedAt: 'updated_at',
		},
	}
);

export default mongoose.model('User', UserSchema);

File: mobile-dev/app/src/main/java/com/hearity_capstone/hearity/data/model/UserModel.kt
package com.hearity_capstone.hearity.data.model

import com.google.gson.annotations.SerializedName

data class UserModel(
    @SerializedName("id") val id: String,
    @SerializedName("first_name") val firstName: String,
    @SerializedName("last_name") val lastName: String,
    @SerializedName("email") val email: String,
    @SerializedName("phone_number") val phoneNumber: String,
    @SerializedName("gender") val gender: String,
    @SerializedName("address") val address: String,
    @SerializedName("city") val city: String,
)
